### PR TITLE
meta-lxatac-bsp: lxatac-factory-data: order service before udev

### PR DESCRIPTION
**Pull request by hnez**
This is an altternative to #107. Since both approaches are valid and have their benefits and drawbacks I made this a competing PR.

Order the factory data collecting service before the udev systemd service to make sure that the `.link` file it creates exists before udev starts looking for rules.

This should fix an issue we have that results in the `tac-bridge` interface having the wrong MAC address on some boots.

In contrast to #107 it does so without triggering udev to reload its rules.

### CODE REVIEW

File: meta-lxatac-bsp/recipes-core/lxatac-factory-data/files/lxatac-factory-data.service [hunk 1]
```diff
@@ -1,10 +1,13 @@
 [Unit]
 Description=Gather and Distribute Factory Data
-Before=NetworkManager.service
+DefaultDependencies=no
+After=systemd-remount-fs.service
+Before=systemd-udevd.service
+RequiresMountsFor=/run
```
Comment (michaelolbrich) [inline meta-lxatac-bsp/recipes-core/lxatac-factory-data/files/lxatac-factory-data.service]: `/run` is always mounted before any service starts, so this line is unnecessary.

File: meta-lxatac-bsp/recipes-core/lxatac-factory-data/files/lxatac-factory-data.sh [hunk 2]
```diff
@@ -67,4 +65,3 @@ OriginalName=tac-bridge
 MACAddress=${MAC_BRIDGE}
 MACAddressPolicy=none
 EOF
-
```
Comment (michaelolbrich) [inline meta-lxatac-bsp/recipes-core/lxatac-factory-data/files/lxatac-factory-data.sh:70]: Unrelated change.

File: meta-lxatac-bsp/recipes-core/lxatac-factory-data/files/lxatac-factory-data.sh [hunk 3]
```diff
@@ -11,14 +11,12 @@ DST_LINK_FILE_BASE="/run/systemd/network/"
...
-# (The transient hostname is passed via systemd.hostname= commandline)
-# --------------------------------------------------------------------
+# If no hostname is set yet persist the one we currently have
+# -----------------------------------------------------------
 
 if [[ ! -e /etc/hostname ]]
 then
-    hostname="$(hostnamectl --transient hostname)"
-    hostnamectl --static hostname "${hostname}"
+    hostname > /etc/hostname
```
Comment (michaelolbrich) [inline meta-lxatac-bsp/recipes-core/lxatac-factory-data/files/lxatac-factory-data.sh:21]: This change deserves to be mentioned in the commit message.

**Review by michaelolbrich — CHANGES_REQUESTED** [review]

File: meta-lxatac-bsp/recipes-core/lxatac-factory-data/files/lxatac-factory-data.service (hunk 1 above)
Comment (hnez) [reply meta-lxatac-bsp/recipes-core/lxatac-factory-data/files/lxatac-factory-data.service]: I did a quick `rg RequiresMountsFor` before adding that to check if other service files in the wild had it and there were some.
But if we do not need it that's even better.

Dropped the `RequiresMountsFor=`.

File: meta-lxatac-bsp/recipes-core/lxatac-factory-data/files/lxatac-factory-data.sh (hunk 2 above)
Comment (hnez) [reply meta-lxatac-bsp/recipes-core/lxatac-factory-data/files/lxatac-factory-data.sh:70]: Split into a separate commit.

File: meta-lxatac-bsp/recipes-core/lxatac-factory-data/files/lxatac-factory-data.sh (hunk 3 above)
Comment (hnez) [reply meta-lxatac-bsp/recipes-core/lxatac-factory-data/files/lxatac-factory-data.sh:21]: Split into a separate commit with some info in the commit message.
Also added more information in the comment.

**Review by michaelolbrich — APPROVED** [review]
The correct spelling is "D-Bus" but in the commit message it does not really matter.
The rest looks good
